File: Output/codestral-mamba-latest/regression.py
# Import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# Generate sample house price dataset
np.random.seed(123)
data = pd.DataFrame({
    'SquareFeet': np.random.uniform(800, 3500, 100),  # Random house sizes
    'Bedrooms': np.random.randint(1, 6, 100),  # Random bedroom counts
    'Age': np.random.randint(1, 51, 100)  # Random house ages
})

# Apply price formula
data['Price'] = data['SquareFeet'] * 100 + data['Bedrooms'] * 5000 - data['Age'] * 300 + np.random.normal(0, 15000, 100)

# Split data into training (70%) and testing (30%)
train_data, test_data = train_test_split(data, train_size=0.7, random_state=123)

# Train linear regression model
model = LinearRegression()
model.fit(train_data[['SquareFeet', 'Bedrooms', 'Age']], train_data['Price'])

# Display model summary
print("Model Summary:")
print(f"Coefficients: {model.coef_}")
print(f"Intercept: {model.intercept_}")

# Predict house prices on test set
test_data['PredictedPrice'] = model.predict(test_data[['SquareFeet', 'Bedrooms', 'Age']])

# Calculate model accuracy
mse = mean_squared_error(test_data['Price'], test_data['PredictedPrice'])
r2 = r2_score(test_data['Price'], test_data['PredictedPrice'])
print("\nModel Accuracy Metrics:")
print(f"Mean Squared Error: {mse}")
print(f"R-squared: {r2}")

# Scatter plot of actual vs predicted house prices
plt.scatter(test_data['Price'], test_data['PredictedPrice'], color='blue', alpha=0.6)
plt.plot(test_data['Price'], test_data['Price'], color='red', linestyle='dashed')
plt.xlabel('Actual Price ($)')
plt.ylabel('Predicted Price ($)')
plt.title('Actual vs Predicted House Prices')
plt.show()

File: Input/regression.r
# Load necessary libraries
library(ggplot2)
library(caret)

# Generate sample house price dataset
set.seed(123)
data <- data.frame(
  SquareFeet = runif(100, min = 800, max = 3500),  # Random house sizes
  Bedrooms = sample(1:5, 100, replace = TRUE),     # Random bedroom counts
  Age = sample(1:50, 100, replace = TRUE),         # Random house ages
  Price = function(x) {                            # Generate house price
    x$SquareFeet * 100 + x$Bedrooms * 5000 - x$Age * 300 + rnorm(100, mean = 0, sd = 15000)
  }
)

data$Price <- data$Price(data)  # Apply price formula

# Split data into training (70%) and testing (30%)
set.seed(123)
trainIndex <- createDataPartition(data$Price, p = 0.7, list = FALSE)
trainData <- data[trainIndex, ]
testData <- data[-trainIndex, ]

# Train linear regression model
model <- lm(Price ~ SquareFeet + Bedrooms + Age, data = trainData)

# Display model summary
cat("Model Summary:\n")
summary(model)

# Predict house prices on test set
testData$PredictedPrice <- predict(model, newdata = testData)

# Calculate model accuracy
accuracy <- postResample(testData$PredictedPrice, testData$Price)
cat("\nModel Accuracy Metrics:\n")
print(accuracy)

# Scatter plot of actual vs predicted house prices
ggplot(testData, aes(x = Price, y = PredictedPrice)) +
  geom_point(color = "blue", alpha = 0.6) +
  geom_abline(slope = 1, intercept = 0, color = "red", linetype = "dashed") +
  labs(title = "Actual vs Predicted House Prices",
       x = "Actual Price ($)",
       y = "Predicted Price ($)") +
  theme_minimal()
